common/request.go

```diff
@@ -14,6 +14,11 @@ import (
 	"github.com/rs/zerolog"
 )
 
+const (
+	CompositeTypeNone      = "none"
+	CompositeTypeLogsSplit = "logs-split"
```

Comment:
Use separate types between when split is happening due to an error or when it's because of auto-splitting threshold
```suggestion
	CompositeTypeLogsSplitOnError = "logs-split-on-error"
	CompositeTypeLogsSplitProactive = "logs-split-proactive"
```

Instead of setting in executeGetLogsSubRequests, set it in upstreamPostForward_eth_getLogs and upstreamPreForward_eth_getLogs accordingly